core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -866,8 +867,11 @@ private[spark] class TaskSetManager(
         isZombie = true
 
         if (fetchFailed.bmAddress != null) {
-          healthTracker.foreach(_.updateExcludedForFetchFailure(
-            fetchFailed.bmAddress.host, fetchFailed.bmAddress.executorId))
+          fetchFailed.bmAddress match {
+            case el: ExecutorLocation =>
+              healthTracker.foreach(_.updateExcludedForFetchFailure(el.host(), el.executorId()))
+            case _ =>
```

Comment:
Could you add a debug level log message here? For the new feature, good log message is crucial for debugging.

Reply:
Sure, good idea!